### ISSUE TITLE
Can only use code formatting after type checking has finished

### ISSUE THREAD

**Issue by calebmer** [rust-lang/rls-vscode#545]
Currently, VSCode only formats my code _after_ RLS has successfully finished type checking. Prettier in the JavaScript community has gotten me into the pattern of writing really messy code and then letting the formatter take care of it, so waiting until after type checking _successfully_ finishes type checking slows down my workflow. Also means I can’t commit formatted code with type errors unless I use `rust-format` on the CLI.

**Versions:**

VSCode extension: 0.5.4

```
$ rustc -V
rustc 1.35.0-nightly (cb2f34dc6 2019-03-22)

$ cargo -V
cargo 1.35.0-nightly (0e35bd8af 2019-03-13)

$ rls -V
rls 1.35.0 (90f7ab0 2019-03-17)
```

**Comment by dvic** [comment]
I can confirm, I sometimes run `cargo fmt` on the command line manually because the type checking takes too long. Is this something that can be fixed? If so, I'd be happy to make a PR if someone can point me in the right direction.

**Comment by Xanewok** [comment]
We're probably trying to be *too* correct inside the RLS. Technically, it's possible to know which file belongs to which crate target (and hence what is the respective *edition* and whether it overlaps with another crate target) only after the analysis is finished, same goes for the code that may have been generated by build scripts.

**Comment by xplorld** [comment]
Are there any plans to fix that? e.g. File's crate is unlikely to change during breaking, and it is tolerable to have mismatched editions until code is type checked. And for the sake of most common cases, could we manually assign workspace-wide editions so that we do not have to wait?

**Comment by ShaneLillieRAD** [comment]
If anything, could we get a Command to run rustfmt in situations where formatting is either stuck or waiting for the type check? I know ```cargo fmt``` so it might be superfluous, but it would just be nice to have it right in the editor like that.
